<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>goatea vignette • goatea</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="goatea vignette">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">goatea</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/goatea.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mauritsunkel/goatea/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><script src="goatea_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="goatea_files/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="goatea_files/vis-9.1.0/vis-network.min.js"></script><script src="goatea_files/visNetwork-binding-2.1.4/visNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>goatea vignette</h1>
                        <h4 data-toc-skip class="author">Maurits
Unkel</h4>
            
            <h4 data-toc-skip class="date">2025-10-29</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mauritsunkel/goatea/blob/HEAD/vignettes/goatea.Rmd" class="external-link"><code>vignettes/goatea.Rmd</code></a></small>
      <div class="hidden name"><code>goatea.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Geneset Ordinal Association Test Enrichment Analysis (GOATEA)
provides a Shiny interface with interactive visualizations and utility
functions for performing and exploring automated gene set enrichment
analysis using the ‘goat’ package.</p>
<p>GOATEA is designed to support large-scale and user-friendly
enrichment workflows across multiple gene lists and comparisons, with
flexible plotting and output options. Visualizations pre-enrichment
include interactive Volcano and UpSet (overlap) plots. Visualizations
post-enrichment include interactive geneset split term dotplot, geneset
hierarchical clustering treeplot, multi-genelist gene-effectsize
heatmap, enrichment overview gene-geneset heatmap and bottom-up
pathway-like STRING database of protein-protein-interactions network
graph.</p>
<div class="section level3">
<h3 id="why-use-goatea">Why use goatea?<a class="anchor" aria-label="anchor" href="#why-use-goatea"></a>
</h3>
<p>Some enrichment visualization packages are included in Bioconductor,
for instance: enrichplot, vissE, CBNplot. goatea additionally provides
interactive visualizations through an intuitive and customizable Shiny
user interface and allows for multi-genelist comparisons. No
bioinformatic expertise is needed. goatea includes commonly used
visualizations and adds a novel enrichment overview heatmap and offers a
bottom-up pathway-like protein-protein interaction graph with network
statistics.</p>
<p>A number of enrichment analysis packages are already on Bioconductor:
clusterProfiler, fgsea, DOSE and many more. In goatea, goat is
implemented. goat is a novel method for efficient geneset enrichment
analysis. It performs within seconds and results in more identified
significant terms compared to other methods, see the reference. For
researchers with coding expertise, goatea provides an automated analysis
workflow for performing enrichment analysis with goat and obtaining
overview tables and output figures to explore your data.</p>
</div>
<div class="section level3">
<h3 id="how-goatea-works">How goatea works<a class="anchor" aria-label="anchor" href="#how-goatea-works"></a>
</h3>
<p>Genelists are often obtained from transcriptomic and/or proteomic
experiments. These genelist tables have to contain a ‘gene’ column with
NCBI Entrez gene identifiers, a ‘symbol’ column with gene aliases
(optional: can be mapped from their gene IDs), and ‘effectsize’ and
‘pvalue’ columns with measurement outcomes for a performed comparison.
User defined thresholds for effect size and p-values are used to define
gene significance. To get a first look and understanding of your
genelist data, a volcano plot is provided. For multi-genelist data a
UpSet (overlap) plot can additionally be generated to visualize
significant genes between comparisons.</p>
<p>Genesets can be obtained by downloading Gene Ontology organism
specific genesets within goatea or from .gmt files, downloaded for
instance from the Molecular Signatures Database. Genelist genes and
genesets will be filtered and matched, then goat or an older version of
gene set enrichment analysis can be performed. Extensive filtering
options help you to explore only the terms and genes within your
vision.</p>
<p>Post-enrichment visualizations help to identify genes and terms of
interest or relevance for your experimental design. Identified and
selected genes of interest are then used to specifically visualize your
data by zooming in on specific biological or technical aspects.</p>
<p>The steps above can be automated, an example workflow is described in
the rest of this vignette.</p>
<p>As intended for interactive and intuitive usage, run the goatea Shiny
interface.</p>
</div>
</div>
<div class="section level2">
<h2 id="running-goatea-shiny-application">Running goatea: Shiny application<a class="anchor" aria-label="anchor" href="#running-goatea-shiny-application"></a>
</h2>
<p>Simply install the package and its dependencies, then run the script
below.</p>
<p>Note that the goatea color scheme is easily customizable, have fun
creating your own theme!</p>
<p>Click the ‘?’ buttons and hover over UI elements for explanation in
tooltips.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mauritsunkel/goatea" class="external-link">goatea</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## customizable coloring</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  main_bg <span class="op">=</span> <span class="st">"#222222"</span>,</span>
<span>  darker_bg <span class="op">=</span> <span class="st">"#111111"</span>,</span>
<span>  focus <span class="op">=</span> <span class="st">"#32CD32"</span>, </span>
<span>  hover <span class="op">=</span> <span class="st">"#228B22"</span>,</span>
<span>  border <span class="op">=</span> <span class="st">"#555555"</span>,</span>
<span>  text <span class="op">=</span> <span class="st">"#FFFFFF"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## run the goatea Shiny application</span></span>
<span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/shinyApp.html" class="external-link">shinyApp</a></span><span class="op">(</span></span>
<span>  ui <span class="op">=</span> <span class="fu">goatea</span><span class="fu">:::</span><span class="va"><a href="../reference/goatea_ui.html">goatea_ui</a></span>,</span>
<span>  server <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">input</span>, <span class="va">output</span>, <span class="va">session</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">goatea</span><span class="fu">:::</span><span class="fu"><a href="../reference/goatea_server.html">goatea_server</a></span><span class="op">(</span></span>
<span>      <span class="va">input</span>, <span class="va">output</span>, <span class="va">session</span>, </span>
<span>      css_colors <span class="op">=</span> <span class="va">colors</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-goatea-automated-analyses">Running goatea: automated analyses<a class="anchor" aria-label="anchor" href="#running-goatea-automated-analyses"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>goatea is available through Github, I intend for goatea to become
available through Bioconductor.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## GOATEA installation requires the latest version of R and Rtools </span></span>
<span><span class="co">## Rtools is needed for package compilation, to download and install visit: </span></span>
<span><span class="co"># R: https://cran.r-project.org/mirrors.html</span></span>
<span><span class="co"># Rtools: https://cran.r-project.org/bin/windows/Rtools/</span></span>
<span></span>
<span><span class="co"># For the GOATEA development version use: </span></span>
<span><span class="kw">if</span> <span class="op">(</span> <span class="op">!</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://pak.r-lib.org/" class="external-link">"pak"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'pak'</span><span class="op">)</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">'mauritsunkel/goatea'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## When GOATEA is on Bioconductor use: </span></span>
<span><span class="co"># pak::pkg_install('goatea', dependencies = TRUE, upgrade = TRUE)</span></span>
<span></span>
<span><span class="co">## goatea organism (taxid) genome wide annotation packages (org.Xx.eg.dg)</span></span>
<span><span class="co">## goatea requires at least one of the following available organism packages:</span></span>
<span><span class="co"># Human (9606)--------: org.Hs.eg.db</span></span>
<span><span class="co"># Mouse (10090)-------: org.Mm.eg.db</span></span>
<span><span class="co"># Fruit Fly (7227)----: org.Dm.eg.db</span></span>
<span><span class="co"># Rhesus monkey (9544): org.Mmu.eg.db</span></span>
<span><span class="co"># Rat (10116)---------: org.Rn.eg.db</span></span>
<span><span class="co"># Worm (6239)---------: org.Ce.eg.db</span></span>
<span><span class="co"># Chimpanzee (9598)---: org.Pt.eg.db</span></span>
<span><span class="co"># Zebrafish (7955)----: org.Dr.eg.db</span></span>
<span><span class="kw">if</span> <span class="op">(</span> <span class="op">!</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://pak.r-lib.org/" class="external-link">"pak"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'pak'</span><span class="op">)</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"org.Hs.eg.db"</span>, </span>
<span>  <span class="st">"org.Mm.eg.db"</span>, </span>
<span>  <span class="st">"org.Dm.eg.db"</span>, </span>
<span>  <span class="st">"org.Mmu.eg.db"</span>, </span>
<span>  <span class="st">"org.Rn.eg.db"</span>, </span>
<span>  <span class="st">"org.Ce.eg.db"</span>, </span>
<span>  <span class="st">"org.Pt.eg.db"</span>, </span>
<span>  <span class="st">"org.Dr.eg.db"</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Optional: add gene descriptions to exported tables, install annotables:</span></span>
<span><span class="co"># available only for: human, mouse, rat, worm, fruit fly, Rhesus Monkey</span></span>
<span><span class="kw">if</span> <span class="op">(</span> <span class="op">!</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://pak.r-lib.org/" class="external-link">"pak"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'pak'</span><span class="op">)</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pkg_install.html" class="external-link">pkg_install</a></span><span class="op">(</span><span class="st">'stephenturner/remotes'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="initialization">Initialization<a class="anchor" aria-label="anchor" href="#initialization"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load goatea package library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mauritsunkel/goatea" class="external-link">goatea</a></span><span class="op">)</span></span></code></pre></div>
<p>Set parameter values used throughout goatea.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set chosen organism taxid (for genome wide annotation package: org.Hs.eg.db)</span></span>
<span><span class="va">taxid</span> <span class="op">&lt;-</span> <span class="fl">9606</span> <span class="co"># Human (Homo Sapiens)</span></span>
<span></span>
<span><span class="co"># set and create output folder</span></span>
<span><span class="va">outdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">outdir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in dir.create(outdir, recursive = TRUE):</span></span>
<span><span class="co">#&gt; 'C:\Users\Home\AppData\Local\Temp\RtmpCwPAku' already exists</span></span>
<span></span>
<span><span class="co"># set significance thresholds</span></span>
<span><span class="va">p_value_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">absolute_effectsize_threshold</span> <span class="op">&lt;-</span> <span class="fl">1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-genelists">Loading genelists<a class="anchor" aria-label="anchor" href="#loading-genelists"></a>
</h3>
<p>Transcriptomic and/or proteomic experiments often end up in comparing
groups through differential gene expression or something similar. There,
lists of genes with measurement weights and statistical probability
values are obtained. Genelists have to be in .csv/.tsv/.xlsx format.
Genelist tables contain a ‘gene’ column with NCBI Entrez gene
identifiers, a ‘symbol’ column with gene aliases (optional: can be
mapped from their gene IDs), and ‘effectsize’ and ‘pvalue’ columns with
measurement outcomes for a performed comparison. Here, goat manuscript
example data will be used, and expected use through uploading your own
file is shown.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># GOAT paper data: download to a specific folder, tempdir() by default</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">goat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/goat/man/download_goat_manuscript_data.html" class="external-link">download_goat_manuscript_data</a></span><span class="op">(</span>output_dir <span class="op">=</span> <span class="va">outdir</span><span class="op">)</span></span>
<span><span class="co">#&gt; downloading https://github.com/ftwkoopmans/goat/raw/main/analyses/goat_manuscript_datasets.rda ...</span></span>
<span><span class="co">#&gt; downloaded data was stored at: C:\Users\Home\AppData\Local\Temp\RtmpCwPAku/goat_manuscript_datasets.rda</span></span>
<span><span class="co"># We select datasets from Colameo 2021 for our automated workflow.</span></span>
<span><span class="co"># Thi is RNA-seq with matched mass-spectrometry data. </span></span>
<span><span class="co"># Rat data is mapped to HUman NCBI Entrez gene identifies with the goat package.</span></span>
<span><span class="va">genelists</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  rna <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">`Colameo 2021:RNA-seq:PMID34396684`</span>,</span>
<span>  ms <span class="op">=</span> <span class="va">data</span><span class="op">$</span><span class="va">`Colameo 2021:mass-spec:PMID34396684`</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genelists</span><span class="op">[[</span><span class="st">'rna'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">#&gt;   symbol log2fc effectsize    pvalue pvalue_adjust hgnc_id hgnc_symbol entrez_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TP53I…   3.03       3.03 1.75<span style="color: #949494;">e</span><span style="color: #BB0000;">-106</span>     2.50<span style="color: #949494;">e</span><span style="color: #BB0000;">-102</span> HGNC:1… TP53INP2        <span style="text-decoration: underline;">58</span>476</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> RAB13    3.12       3.12 9.13<span style="color: #949494;">e</span><span style="color: #BB0000;">- 87</span>     2.39<span style="color: #949494;">e</span><span style="color: #BB0000;">- 83</span> HGNC:9… RAB13            <span style="text-decoration: underline;">5</span>872</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> B2M      2.85       2.85 1.00<span style="color: #949494;">e</span><span style="color: #BB0000;">- 86</span>     2.39<span style="color: #949494;">e</span><span style="color: #BB0000;">- 83</span> HGNC:9… B2M               567</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> METRN    2.55       2.55 4.12<span style="color: #949494;">e</span><span style="color: #BB0000;">- 86</span>     8.43<span style="color: #949494;">e</span><span style="color: #BB0000;">- 83</span> HGNC:1… METRN           <span style="text-decoration: underline;">79</span>006</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> WASF3    2.70       2.70 1.37<span style="color: #949494;">e</span><span style="color: #BB0000;">- 79</span>     2.19<span style="color: #949494;">e</span><span style="color: #BB0000;">- 76</span> HGNC:1… WASF3           <span style="text-decoration: underline;">10</span>810</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> NDRG2    2.81       2.81 7.30<span style="color: #949494;">e</span><span style="color: #BB0000;">- 79</span>     1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">- 75</span> HGNC:1… NDRG2           <span style="text-decoration: underline;">57</span>447</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: gene &lt;int&gt;, signif &lt;lgl&gt;</span></span></span></code></pre></div>
<p>To load your own data follow this alternative example.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set the filepath to your own data files </span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>,<span class="st">'example_genelist.csv'</span>, package <span class="op">=</span> <span class="st">'goatea'</span><span class="op">)</span></span>
<span><span class="co"># Check file formatting, see function documentation for additional parameters </span></span>
<span><span class="va">genelist</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/read_validate_genelist.html">read_validate_genelist</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">file</span><span class="op">)</span></span>
<span><span class="co">#&gt; Checking file format...</span></span>
<span><span class="co"># genelists &lt;- list(</span></span>
<span>  <span class="co"># csv = genelist</span></span>
<span><span class="co"># )</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genelist</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   symbol   gene pvalue effectsize signif</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> gene_45 <span style="text-decoration: underline;">11</span>023 0.123         5   FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> gene_12 <span style="text-decoration: underline;">12</span>763 0.435         5   FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> gene_34 <span style="text-decoration: underline;">16</span>847 0.435         5   FALSE </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> gene_83 <span style="text-decoration: underline;">12</span>069 0.014<span style="text-decoration: underline;">8</span>       -<span style="color: #BB0000;">4.9</span> TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> gene_14 <span style="text-decoration: underline;">17</span>454 0.015<span style="text-decoration: underline;">1</span>        4.8 TRUE  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> gene_96 <span style="text-decoration: underline;">12</span>308 0.375         4.8 FALSE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="loading-genesets">Loading genesets<a class="anchor" aria-label="anchor" href="#loading-genesets"></a>
</h3>
<p>Genesets are terms of biological aspects associated with a set of
genes. The Gene Ontology (GO) and Molecular Signatures databases
(MSigDB) are commonly used repositories containing organism specific
genesets. goat provides GO genesets for download, also to supply a .gmt
file containing genesets downloaded from MSigDB for instance.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesets</span> <span class="op">&lt;-</span> <span class="fu">goat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/goat/man/load_genesets_go_bioconductor.html" class="external-link">load_genesets_go_bioconductor</a></span><span class="op">(</span>taxid <span class="op">=</span> <span class="va">taxid</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; load_genesets_go_bioconductor(): data version = org.Hs.eg.db - Homo sapiens - 2025-02-06</span></span>
<span></span>
<span><span class="co">## optionally: save and load from .rda to speed up after downloading</span></span>
<span><span class="co"># save(genesets, file = file.path(outdir, 'genesets.rda'))</span></span>
<span><span class="co"># load(file.path(outdir, 'genesets.rda')) # loading .rda to variable: genesets</span></span>
<span></span>
<span><span class="co">## alternatively: load via .gmt file (downloaded from e.g.: MSigDB)</span></span>
<span><span class="co"># then use: goat::load_genesets_gmtfile()</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genesets</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   source source_version                       id    name  parent_id genes ngenes</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&lt;ch&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lis&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> GO_BP  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… mito…       <span style="color: #949494;">[1]</span>  <span style="color: #949494;">[30]</span>     30</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> GO_MF  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… alph…       <span style="color: #949494;">[1]</span>   <span style="color: #949494;">[3]</span>      3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> GO_BP  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… sing…       <span style="color: #949494;">[1]</span>  <span style="color: #949494;">[14]</span>     14</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> GO_MF  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… sing…       <span style="color: #949494;">[1]</span>  <span style="color: #949494;">[11]</span>     11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> GO_CC  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… phos…       <span style="color: #949494;">[2]</span>   <span style="color: #949494;">[4]</span>      4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> GO_MF  org.Hs.eg.db - Homo sapiens - 2025-… GO:0… lact…       <span style="color: #949494;">[1]</span>   <span style="color: #949494;">[1]</span>      1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="goat-enrichment-analysis">GOAT Enrichment Analysis<a class="anchor" aria-label="anchor" href="#goat-enrichment-analysis"></a>
</h3>
<p>Gene significance is set per genelist by the user-defined thresholds.
The genelist is then ordered by effect size. For this example, if there
are more genes than the amount of precomputed nulldistributions goat
uses for its enrichment testing the amount of genes is limited to that
amount. If you do want to use all genes, set the method to
‘goat_bootstrap’.</p>
<p>Genelist genes and genesets will be filtered and matched, then goat
or an older version of gene set enrichment analysis can be performed.
Too small or too large genesets can be irrelevant or mess up testing
statistics, leaving parameter values to defaults is a good starting
point.</p>
<p>Geneset enrichment with goat is by default performed on ‘effectsize’.
The user can also go for ‘pvalue’ or a specific direction by gene
effectsize sign or absolute values. Besides the goat method, older
methods are also implemented to enable easy comparison.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered_genesets_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">enrichment_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">name</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genelists</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">genelist</span> <span class="op">&lt;-</span> <span class="va">genelists</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="co">## set signif column by given p-value- and effectsize significance thresholds</span></span>
<span>  <span class="va">genelist</span><span class="op">$</span><span class="va">signif</span> <span class="op">&lt;-</span> <span class="va">genelist</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;=</span> <span class="va">p_value_threshold</span> <span class="op">&amp;</span> </span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">genelist</span><span class="op">$</span><span class="va">effectsize</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">absolute_effectsize_threshold</span></span>
<span>  <span class="co"># order genelist by effectsize </span></span>
<span>  <span class="va">genelist</span> <span class="op">&lt;-</span> <span class="va">genelist</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">genelist</span><span class="op">$</span><span class="va">effectsize</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,<span class="op">]</span></span>
<span>  <span class="co">## keep max n genes for regular goat method</span></span>
<span>  <span class="co"># optionally: remove this line and use goat_bootstrap method </span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">genelist</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu">goat</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/goat/man/goat_nulldistributions.html" class="external-link">goat_nulldistributions</a></span><span class="op">$</span><span class="va">N</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">genelist</span> <span class="op">&lt;-</span> <span class="va">genelist</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu">goat</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/goat/man/goat_nulldistributions.html" class="external-link">goat_nulldistributions</a></span><span class="op">$</span><span class="va">N</span><span class="op">)</span>,<span class="op">]</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="co"># update genelist</span></span>
<span>  <span class="va">genelists</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">genelist</span></span>
<span>  </span>
<span>  <span class="co">## filter genesets </span></span>
<span>  <span class="va">filtered_genesets</span> <span class="op">&lt;-</span> <span class="fu">goat</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/goat/man/filter_genesets.html" class="external-link">filter_genesets</a></span><span class="op">(</span></span>
<span>    genesets <span class="op">=</span> <span class="va">genesets</span>,</span>
<span>    genelist <span class="op">=</span> <span class="va">genelist</span>,</span>
<span>    min_overlap <span class="op">=</span> <span class="fl">10L</span>,</span>
<span>    max_overlap <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    max_overlap_fraction <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    min_signif <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    max_size <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>    dedupe <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="va">filtered_genesets_list</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">filtered_genesets</span></span>
<span></span>
<span>  <span class="co">## run enrichment </span></span>
<span>  <span class="va">enrichment_results</span><span class="op">[[</span><span class="va">name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/run_geneset_enrichment.html">run_geneset_enrichment</a></span><span class="op">(</span></span>
<span>    genesets <span class="op">=</span> <span class="va">filtered_genesets</span>,</span>
<span>    genelist <span class="op">=</span> <span class="va">genelist</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"goat"</span>,</span>
<span>    score_type <span class="op">=</span> <span class="st">"effectsize"</span>,</span>
<span>    padj_method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>    padj_sources <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    padj_cutoff <span class="op">=</span> <span class="va">p_value_threshold</span>,</span>
<span>    padj_min_signifgenes <span class="op">=</span> <span class="fl">0L</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 17</span></span></span>
<span><span class="co">#&gt;   source source_version  id    name  parent_id ngenes_input ngenes ngenes_signif</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&lt;ch&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> GO_CC  org.Hs.eg.db -… GO:0… cyto…       <span style="color: #949494;">[2]</span>          118     97            76</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> GO_CC  org.Hs.eg.db -… GO:0… cyto…       <span style="color: #949494;">[2]</span>         <span style="text-decoration: underline;">5</span>530   <span style="text-decoration: underline;">4</span>335           687</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> GO_BP  org.Hs.eg.db -… GO:0… cyto…       <span style="color: #949494;">[1]</span>          169    147            83</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> GO_MF  org.Hs.eg.db -… GO:0… tran…       <span style="color: #949494;">[1]</span>         <span style="text-decoration: underline;">1</span>320    394           105</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> GO_MF  org.Hs.eg.db -… GO:0… stru…       <span style="color: #949494;">[1]</span>          175    144            78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> GO_MF  org.Hs.eg.db -… GO:0… sign…       <span style="color: #949494;">[1]</span>         <span style="text-decoration: underline;">1</span>554    518           123</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 9 more variables: genes &lt;list&lt;int&gt;&gt;, genes_signif &lt;list&gt;, score_type &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   pvalue &lt;dbl&gt;, zscore &lt;dbl&gt;, pvalue_adjust &lt;dbl&gt;, signif &lt;lgl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   score_oddsratio &lt;dbl&gt;, symbol &lt;list&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="genes-overview-table">Genes overview table<a class="anchor" aria-label="anchor" href="#genes-overview-table"></a>
</h3>
<p>Here, the genelists are formatted into a gene-based overview table.
If there are multiple genelists, for all genelists, each gene and their
p-values, effect sizes, effect size percentages, gene set ratios and
overlapping significant genes will be shown.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes_overview</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/run_genelists_overlap.html">run_genelists_overlap</a></span><span class="op">(</span><span class="va">genelists</span><span class="op">)</span></span>
<span><span class="va">genes_overview</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/calculate_geneSetRatio.html">calculate_geneSetRatio</a></span><span class="op">(</span></span>
<span>  <span class="va">enrichment_results</span>, </span>
<span>  <span class="va">genes_overview</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genes_overview</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 13</span></span></span>
<span><span class="co">#&gt;     gene symbol   rna_efsi rna_pval ms_efsi ms_pval rna_perc ms_perc</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>   <span style="text-decoration: underline;">3</span>036 HAS1        -<span style="color: #BB0000;">6.74</span> 3.94<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>    -<span style="color: #BB0000;">99.1</span>      <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>  <span style="text-decoration: underline;">80</span>736 SLC44A4     -<span style="color: #BB0000;">4.39</span> 2.22<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>    -<span style="color: #BB0000;">95.2</span>      <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>   <span style="text-decoration: underline;">4</span>509 MT-ATP8      3.91 4.90<span style="color: #949494;">e</span><span style="color: #BB0000;">-37</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="text-decoration: underline;">1</span>399.       <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> <span style="text-decoration: underline;">284</span>613 CYB561D1    -<span style="color: #BB0000;">3.81</span> 2.67<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>    -<span style="color: #BB0000;">92.9</span>      <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>   <span style="text-decoration: underline;">5</span>730 PTGDS       -<span style="color: #BB0000;">3.62</span> 9.69<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>    -<span style="color: #BB0000;">91.9</span>      <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span>   <span style="text-decoration: underline;">2</span>830 GPR6        -<span style="color: #BB0000;">3.41</span> 2.85<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>    -<span style="color: #BB0000;">90.6</span>      <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5 more variables: genelist_overlap &lt;chr&gt;, rna_geneSetRatio &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   rna_SS_geneSetRatio &lt;dbl&gt;, ms_geneSetRatio &lt;dbl&gt;, ms_SS_geneSetRatio &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="enrichment-searching-and-filtering">Enrichment searching and filtering<a class="anchor" aria-label="anchor" href="#enrichment-searching-and-filtering"></a>
</h3>
<p>You can filter enrichment by querying the term names and by
thresholding the numerical features. In this way you can explore
relevant biological aspects and/or filter out technicalities. After
filtering, use the filtered enrichment as input with the visualizations
for zooming in on specific terms and their genes and interactions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtered_enrichment</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/filter_enrichment.html">filter_enrichment</a></span><span class="op">(</span></span>
<span>  df <span class="op">=</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  terms_query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"synapse"</span>, <span class="st">"synaptic"</span><span class="op">)</span>, </span>
<span>  terms_antiquery <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"asymmetric"</span><span class="op">)</span>,</span>
<span>  min_ngenes <span class="op">=</span> <span class="fl">100</span>, </span>
<span>  min_ngenes_signif <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "synaptic membrane"                   </span></span>
<span><span class="co">#&gt; [2] "postsynaptic membrane"               </span></span>
<span><span class="co">#&gt; [3] "postsynaptic density"                </span></span>
<span><span class="co">#&gt; [4] "postsynaptic specialization membrane"</span></span>
<span><span class="co">#&gt; [5] "postsynaptic specialization"         </span></span>
<span><span class="co">#&gt; [6] "trans-synaptic signaling"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="gene-selection">Gene selection<a class="anchor" aria-label="anchor" href="#gene-selection"></a>
</h3>
<p>In the goatea app, the visualizations are used to select specific
genes to pass to next visualizations in order to zoom in on specific
terms or groups of genes. Here we simply take the significant genes of
the first filtered enrichment to show some visualizations that use gene
selections.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_genes</span> <span class="op">&lt;-</span> <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">genes_signif</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plotting">Plotting<a class="anchor" aria-label="anchor" href="#plotting"></a>
</h2>
<div class="section level3">
<h3 id="genelist-plotting">Genelist plotting<a class="anchor" aria-label="anchor" href="#genelist-plotting"></a>
</h3>
<div class="section level4">
<h4 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h4>
<p>A Volcano plot is commonly used to have a visual representation of
the genelist. It shows an overview of genes as points with effectsize vs
p-value and significance thresholds.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_EnhancedVolcano.html">plot_EnhancedVolcano</a></span><span class="op">(</span></span>
<span>  genelist <span class="op">=</span> <span class="va">genelists</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  effectsize_threshold <span class="op">=</span> <span class="va">absolute_effectsize_threshold</span>,</span>
<span>  pvalue_threshold <span class="op">=</span> <span class="va">p_value_threshold</span>,</span>
<span>  background_color <span class="op">=</span> <span class="st">'white'</span>,</span>
<span>  foreground_color <span class="op">=</span> <span class="st">'black'</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">EnhancedVolcano</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">EnhancedVolcano</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="co">#&gt; Coordinate system already present.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Adding new coordinate system, which will replace the existing one.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 2229 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="goatea_files/figure-html/genelist_volcano-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="upset-overlap-plot">UpSet overlap plot<a class="anchor" aria-label="anchor" href="#upset-overlap-plot"></a>
</h4>
<p>If multiple genelists are loaded, their overlap of significant genes
is shown through an UpSet plot. These show the total number of genes in
a genelist and the number of overlapping significant genes for each
intersection.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genelists</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_genelists_overlap_upsetjs.html">plot_genelists_overlap_upsetjs</a></span><span class="op">(</span></span>
<span>    genelists <span class="op">=</span> <span class="va">genelists</span>, </span>
<span>    mode <span class="op">=</span> <span class="st">'distinct'</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="post-enrichment-plotting">Post-enrichment plotting<a class="anchor" aria-label="anchor" href="#post-enrichment-plotting"></a>
</h3>
<div class="section level4">
<h4 id="tree-plot">Tree plot<a class="anchor" aria-label="anchor" href="#tree-plot"></a>
</h4>
<p>The tree plot shows a clustered hierarchical split between terms
based on their term similarity. Similarity is based on semantics of the
term. Terms are clustered together by semantics, meaning words that look
or mean alike. A summary of the clustered semantics is given to the
right of the vertical lines for each cluster.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter to keep a single geneset source</span></span>
<span><span class="va">single_source_enrichment</span> <span class="op">&lt;-</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">source</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_termtree.html">plot_termtree</a></span><span class="op">(</span></span>
<span>  <span class="va">single_source_enrichment</span>, </span>
<span>  Nterms <span class="op">=</span> <span class="fl">20</span>, </span>
<span>  Nwords <span class="op">=</span> <span class="fl">5</span>, </span>
<span>  Nclusters <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: `aes_()` was deprecated in ggplot2 3.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use tidy evaluation idioms with `aes()`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">enrichplot</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at</span></span>
<span><span class="co">#&gt;   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/GuangchuangYu/enrichplot/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<p><img src="goatea_files/figure-html/tree_plot-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="term-split-dotplot">Term-split dotplot<a class="anchor" aria-label="anchor" href="#term-split-dotplot"></a>
</h4>
<p>The dotplot shows terms split by up- and down regulation. The
regulation is defined by the sign of the effect sizes of the genes in
the genelists. The gene ratio shows the percentage of genes of the term
that were matched with the genes of the genelist.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># filter to keep a single geneset source</span></span>
<span><span class="va">single_source_enrichment</span> <span class="op">&lt;-</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">source</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_splitdot.html">plot_splitdot</a></span><span class="op">(</span><span class="va">single_source_enrichment</span>, topN <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="goatea_files/figure-html/term_split_dotplot-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="gene-genelist-heatmap">Gene-genelist heatmap<a class="anchor" aria-label="anchor" href="#gene-genelist-heatmap"></a>
</h4>
<p>This heatmap shows the selected genes for each genelist to compare
their effect sizes and p-values as a simple overview.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selected_symbols</span> <span class="op">&lt;-</span> <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">symbol</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span></span>
<span>  <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">genes_signif</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_gene_effectsize_ComplexHeatmap.html">plot_gene_effectsize_ComplexHeatmap</a></span><span class="op">(</span></span>
<span>  genes <span class="op">=</span> <span class="va">selected_symbols</span>, </span>
<span>  genes_overview <span class="op">=</span> <span class="va">genes_overview</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="goatea_files/figure-html/gene_genelist_heatmap-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="gene-term-heatmap">Gene-term heatmap<a class="anchor" aria-label="anchor" href="#gene-term-heatmap"></a>
</h4>
<p>This heatmap serves as an overview of the enrichment result, combined
with all genelists information. Use the search functionality and pass
the resulting filtered enrichment to zoom in on relevant biological
aspects and focus on filtered aspects of the enrichment for exploration.
In the goatea app this heatmap is also used to select genes/terms from
identified aspects of interest.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_ComplexHeatmap.html">plot_ComplexHeatmap</a></span><span class="op">(</span></span>
<span>  enrichment_result <span class="op">=</span> <span class="va">enrichment_results</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  genelist <span class="op">=</span> <span class="va">genelists</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>  n_cluster <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  n_top_terms <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  n_top_genes <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  genelist_overlap <span class="op">=</span> <span class="va">genes_overview</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Following `at` are removed: NA, because no color was defined for them.</span></span>
<span><span class="co">#&gt; Following `at` are removed: NA, because no color was defined for them.</span></span>
<span><span class="co">#&gt; Following `at` are removed: NA, because no color was defined for them.</span></span></code></pre></div>
<p><img src="goatea_files/figure-html/gene_term_heatmap-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="protein-protein-interaction-network">Protein-protein interaction network<a class="anchor" aria-label="anchor" href="#protein-protein-interaction-network"></a>
</h4>
<p>Gene sets miss information of gene-gene interactions, this
visualization can offer bottom-up relationships between genes to gain
insight. The STRING database is used to obtain protein-protein
interaction data by efficient streaming. For the selected genes of
interest, an igraph network is built in the background in order to
obtain network statistics. The igraph is passed to visNetwork for (in
Shiny app) interactive visualization and exploration of the network.</p>
<p>First we select symbols to download ppi data for our chosen
organism.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># selecting 10 symbols of the searchad and significant first enriched term</span></span>
<span><span class="va">selected_symbols</span> <span class="op">&lt;-</span> <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">symbol</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span></span>
<span>  <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">genes</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">filtered_enrichment</span><span class="op">$</span><span class="va">genes_signif</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/assign.html" class="external-link">assign</a></span><span class="op">(</span><span class="st">"ppi_data"</span>, <span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_ppi_data.rda"</span>, package <span class="op">=</span> <span class="st">"goatea"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Normally, we would download STRING database PPI data, yet for this
vignette, we will use example data by default. Try out this code for
yourself to explore with downloaded data.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## this will download the STRING database protein-protein interaction data </span></span>
<span><span class="co">## to your selected folder </span></span>
<span><span class="co"># downloading should only take a couple of seconds</span></span>
<span><span class="co"># downlaod happens only if the files are not found in the specified folder</span></span>
<span><span class="co"># if the download does not work, double check if the STRING database is online</span></span>
<span><span class="va">ppi_data</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/get_string_ppi.html">get_string_ppi</a></span><span class="op">(</span></span>
<span>  aliases <span class="op">=</span> <span class="va">selected_symbols</span>, </span>
<span>  organism <span class="op">=</span> <span class="va">taxid</span>, </span>
<span>  folder <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">outdir</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>I advise to regard this output as a basic and toy visualization, the
goatea in Shiny app version contains node and edge hover information
with all network statistics. Edges can be clicked to view their
interaction scores and information on the STRING website. Nodes are
clustered by STRING interaction scores. The nodes and edges can be
colored by each statistical feature. Nodes can be filtered by each
feature and made into a subnetwork for more detailed exploration.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># creating igraph in order to do network statistics</span></span>
<span><span class="va">ppi_graph</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/get_ppigraph.html">get_ppigraph</a></span><span class="op">(</span>ppi_data <span class="op">=</span> <span class="va">ppi_data</span><span class="op">)</span></span>
<span><span class="co"># converting igraph to visnetwork for (Shiny app) interactive visualization</span></span>
<span><span class="va">vis_network</span> <span class="op">&lt;-</span> <span class="fu">goatea</span><span class="fu">::</span><span class="fu"><a href="../reference/get_visNetwork.html">get_visNetwork</a></span><span class="op">(</span></span>
<span>  ppigraph <span class="op">=</span> <span class="va">ppi_graph</span>, </span>
<span>  genes_overview <span class="op">=</span> <span class="va">genes_overview</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in min(genes_overview[[paste0(sample_name, "_efsi")]], na.rm = TRUE):</span></span>
<span><span class="co">#&gt; no non-missing arguments to min; returning Inf</span></span>
<span><span class="co">#&gt; Warning in max(genes_overview[[paste0(sample_name, "_efsi")]], na.rm = TRUE):</span></span>
<span><span class="co">#&gt; no non-missing arguments to max; returning -Inf</span></span>
<span><span class="co">## showing a basic version of the visNetwork</span></span>
<span><span class="co">## the goatea Shiny app version has full interactive capability </span></span>
<span><span class="fu">visNetwork</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visNetwork.html" class="external-link">visNetwork</a></span><span class="op">(</span></span>
<span>  <span class="va">vis_network</span><span class="op">$</span><span class="va">nodes</span>, </span>
<span>  <span class="va">vis_network</span><span class="op">$</span><span class="va">edges</span>, </span>
<span>  width <span class="op">=</span> <span class="st">"100%"</span>, </span>
<span>  height <span class="op">=</span> <span class="st">"100%"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu">visNetwork</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/visNetwork/man/visPhysics.html" class="external-link">visPhysics</a></span><span class="op">(</span>stabilization <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:100%;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"nodes":{"label":["TP53","EGFR","SOX2","MTOR","BRCA1","MYC"],"cluster":[1,1,1,1,1,1],"degree":[5,5,5,5,5,5],"betweenness":[0,0,0,0,0,0],"closeness":[0.2,0.2,0.2,0.2,0.2,0.2],"knn":[5,5,5,5,5,5],"diversity":[0.9979801712322223,0.9958339678473881,0.9021086969850233,0.9300986772386417,0.9132781884336926,0.9916855551370735],"id":["9606.ENSP00000269305","9606.ENSP00000275493","9606.ENSP00000323588","9606.ENSP00000354558","9606.ENSP00000418960","9606.ENSP00000478887"],"title":["Betweenness: 0","Betweenness: 0","Betweenness: 0","Betweenness: 0","Betweenness: 0","Betweenness: 0"],"color.background":["#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF"],"color.border":["#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF","#E69F00FF"],"gene":[7157,1956,6657,2475,672,4609],"symbol":["TP53","EGFR","SOX2","MTOR","BRCA1","MYC"],"rna_efsi":[0.107227744656282,-0.486016535224295,0.170214438165599,0.0294208357325172,0.382183241887704,0.0429659400067511],"rna_pval":[0.562079177781425,0.000368880362393361,0.181794578502844,0.814526403046293,0.07791381547361489,0.880744523459519],"ms_efsi":[null,2.80761021282852,null,-0.764944087644817,null,null],"ms_pval":[null,0.0136896259973968,null,0.456682015776396,null,null],"rna_perc":[7.715640329232198,-28.60061935330652,12.52257226715217,2.060232665424322,30.33126772831527,3.02296310657606],"ms_perc":[null,600.1238788147658,null,-41.15238251492465,null,null],"genelist_overlap":["","ms","","","",""],"rna_geneSetRatio":[0.3988300983780909,0.2658867322520606,1.794735442701409,0.5583621377293273,1.090135602233449,0.4785961180537092],"rna_SS_geneSetRatio":[6.924882629107981,2.464788732394366,2.934272300469484,0.3521126760563381,5.39906103286385,0.9389671361502347],"ms_geneSetRatio":[0,1.585503963759909,0,1.540203850509626,0,0],"ms_SS_geneSetRatio":[0,1.62052667116813,0,1.688048615800135,0,0]},"edges":{"combined_score":[997,999,883,874,943,802,758,800,875,958,443,419,497,730,999],"from":["9606.ENSP00000269305","9606.ENSP00000269305","9606.ENSP00000269305","9606.ENSP00000269305","9606.ENSP00000269305","9606.ENSP00000275493","9606.ENSP00000275493","9606.ENSP00000275493","9606.ENSP00000275493","9606.ENSP00000323588","9606.ENSP00000323588","9606.ENSP00000323588","9606.ENSP00000354558","9606.ENSP00000354558","9606.ENSP00000418960"],"to":["9606.ENSP00000478887","9606.ENSP00000418960","9606.ENSP00000323588","9606.ENSP00000354558","9606.ENSP00000275493","9606.ENSP00000354558","9606.ENSP00000323588","9606.ENSP00000418960","9606.ENSP00000478887","9606.ENSP00000478887","9606.ENSP00000418960","9606.ENSP00000354558","9606.ENSP00000418960","9606.ENSP00000478887","9606.ENSP00000478887"],"edge_betweenness":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"width":[2.493103448275862,2.5,2.1,2.068965517241379,2.306896551724138,1.820689655172414,1.668965517241379,1.813793103448276,2.072413793103449,2.358620689655173,0.5827586206896552,0.5,0.7689655172413794,1.572413793103448,2.5],"id":["9606.ENSP00000269305_9606.ENSP00000478887","9606.ENSP00000269305_9606.ENSP00000418960","9606.ENSP00000269305_9606.ENSP00000323588","9606.ENSP00000269305_9606.ENSP00000354558","9606.ENSP00000269305_9606.ENSP00000275493","9606.ENSP00000275493_9606.ENSP00000354558","9606.ENSP00000275493_9606.ENSP00000323588","9606.ENSP00000275493_9606.ENSP00000418960","9606.ENSP00000275493_9606.ENSP00000478887","9606.ENSP00000323588_9606.ENSP00000478887","9606.ENSP00000323588_9606.ENSP00000418960","9606.ENSP00000323588_9606.ENSP00000354558","9606.ENSP00000354558_9606.ENSP00000418960","9606.ENSP00000354558_9606.ENSP00000478887","9606.ENSP00000418960_9606.ENSP00000478887"],"title":["STRING score: 997<br>Betweenness: 1<br>","STRING score: 999<br>Betweenness: 1<br>","STRING score: 883<br>Betweenness: 1<br>","STRING score: 874<br>Betweenness: 1<br>","STRING score: 943<br>Betweenness: 1<br>","STRING score: 802<br>Betweenness: 1<br>","STRING score: 758<br>Betweenness: 1<br>","STRING score: 800<br>Betweenness: 1<br>","STRING score: 875<br>Betweenness: 1<br>","STRING score: 958<br>Betweenness: 1<br>","STRING score: 443<br>Betweenness: 1<br>","STRING score: 419<br>Betweenness: 1<br>","STRING score: 497<br>Betweenness: 1<br>","STRING score: 730<br>Betweenness: 1<br>","STRING score: 999<br>Betweenness: 1<br>"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"physics":{"stabilization":false}},"groups":null,"width":"100%","height":"100%","idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div class="section level2">
<h2 id="information">Information<a class="anchor" aria-label="anchor" href="#information"></a>
</h2>
<p>A thank you for your time and effort in using goatea, I hope it may
aid you in exploring your data!</p>
<div class="section level3">
<h3 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h3>
<p>goatea was developed by Maurits Unkel at the Erasmus Medical Center
in the department of Psychiatry. Source code is available through
GitHub: <a href="https://github.com/mauritsunkel/goatea" class="external-link uri">https://github.com/mauritsunkel/goatea</a> Questions
regarding goatea can be sent to: <a href="mailto:mauritsunkel@gmail.com" class="email">mauritsunkel@gmail.com</a></p>
</div>
<div class="section level3">
<h3 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h3>
<p>goatea is licensed under Apache 2.0.</p>
</div>
<div class="section level3">
<h3 id="issues-and-contributions">Issues and contributions<a class="anchor" aria-label="anchor" href="#issues-and-contributions"></a>
</h3>
<p>For issues and contributions, please find the goatea Github page: <a href="https://github.com/mauritsunkel/goatea/issues" class="external-link uri">https://github.com/mauritsunkel/goatea/issues</a></p>
</div>
<div class="section level3">
<h3 id="goat-reference">GOAT reference<a class="anchor" aria-label="anchor" href="#goat-reference"></a>
</h3>
<p>Koopmans, F. GOAT: efficient and robust identification of gene set
enrichment. Commun Biol 7, 744 (2024). <a href="https://doi.org/10.1038/s42003-024-06454-5" class="external-link uri">https://doi.org/10.1038/s42003-024-06454-5</a></p>
</div>
<div class="section level3">
<h3 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.0 (2025-04-11 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 11 x64 (build 26200)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt;   LAPACK version 3.12.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Amsterdam</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] goatea_0.99.9    dplyr_1.1.4      BiocStyle_2.37.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3               rstudioapi_0.17.1               </span></span>
<span><span class="co">#&gt;   [3] jsonlite_2.0.0                   shape_1.4.6.1                   </span></span>
<span><span class="co">#&gt;   [5] magrittr_2.0.4                   magick_2.9.0                    </span></span>
<span><span class="co">#&gt;   [7] shinyjqui_0.4.1                  ggtangle_0.0.7                  </span></span>
<span><span class="co">#&gt;   [9] farver_2.1.2                     rmarkdown_2.30                  </span></span>
<span><span class="co">#&gt;  [11] GlobalOptions_0.1.2              fs_1.6.6                        </span></span>
<span><span class="co">#&gt;  [13] ragg_1.4.0                       vctrs_0.6.5                     </span></span>
<span><span class="co">#&gt;  [15] memoise_2.0.1                    ggtree_3.99.0                   </span></span>
<span><span class="co">#&gt;  [17] htmltools_0.5.8.1                gridGraphics_0.5-1              </span></span>
<span><span class="co">#&gt;  [19] sass_0.4.10                      bslib_0.9.0                     </span></span>
<span><span class="co">#&gt;  [21] htmlwidgets_1.6.4                desc_1.4.3                      </span></span>
<span><span class="co">#&gt;  [23] fontawesome_0.5.3                plyr_1.8.9                      </span></span>
<span><span class="co">#&gt;  [25] plotly_4.11.0                    cachem_1.1.0                    </span></span>
<span><span class="co">#&gt;  [27] uuid_1.2-1                       igraph_2.2.0                    </span></span>
<span><span class="co">#&gt;  [29] mime_0.13                        lifecycle_1.0.4                 </span></span>
<span><span class="co">#&gt;  [31] iterators_1.0.14                 pkgconfig_2.0.3                 </span></span>
<span><span class="co">#&gt;  [33] Matrix_1.7-4                     R6_2.6.1                        </span></span>
<span><span class="co">#&gt;  [35] InteractiveComplexHeatmap_1.17.0 fastmap_1.2.0                   </span></span>
<span><span class="co">#&gt;  [37] shiny_1.11.1                     clue_0.3-66                     </span></span>
<span><span class="co">#&gt;  [39] digest_0.6.37                    aplot_0.2.9                     </span></span>
<span><span class="co">#&gt;  [41] enrichplot_1.29.2                ggnewscale_0.5.2                </span></span>
<span><span class="co">#&gt;  [43] colorspace_2.1-2                 patchwork_1.3.2                 </span></span>
<span><span class="co">#&gt;  [45] AnnotationDbi_1.71.0             S4Vectors_0.47.0                </span></span>
<span><span class="co">#&gt;  [47] textshaping_1.0.3                RSQLite_2.4.3                   </span></span>
<span><span class="co">#&gt;  [49] org.Hs.eg.db_3.21.0              labeling_0.4.3                  </span></span>
<span><span class="co">#&gt;  [51] httr_1.4.7                       compiler_4.5.0                  </span></span>
<span><span class="co">#&gt;  [53] withr_3.0.2                      bit64_4.6.0-1                   </span></span>
<span><span class="co">#&gt;  [55] doParallel_1.0.17                S7_0.2.0                        </span></span>
<span><span class="co">#&gt;  [57] BiocParallel_1.43.4              DBI_1.2.3                       </span></span>
<span><span class="co">#&gt;  [59] R.utils_2.13.0                   rappdirs_0.3.3                  </span></span>
<span><span class="co">#&gt;  [61] rjson_0.2.23                     tools_4.5.0                     </span></span>
<span><span class="co">#&gt;  [63] ape_5.8-1                        zip_2.3.3                       </span></span>
<span><span class="co">#&gt;  [65] httpuv_1.6.16                    R.oo_1.27.1                     </span></span>
<span><span class="co">#&gt;  [67] glue_1.8.0                       nlme_3.1-168                    </span></span>
<span><span class="co">#&gt;  [69] GOSemSim_2.35.1                  promises_1.3.3                  </span></span>
<span><span class="co">#&gt;  [71] grid_4.5.0                       cluster_2.1.8.1                 </span></span>
<span><span class="co">#&gt;  [73] reshape2_1.4.4                   fgsea_1.35.6                    </span></span>
<span><span class="co">#&gt;  [75] generics_0.1.4                   gtable_0.3.6                    </span></span>
<span><span class="co">#&gt;  [77] tzdb_0.5.0                       R.methodsS3_1.8.2               </span></span>
<span><span class="co">#&gt;  [79] tidyr_1.3.1                      data.table_1.17.8               </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.6                       xml2_1.4.0                      </span></span>
<span><span class="co">#&gt;  [83] XVector_0.49.0                   BiocGenerics_0.55.0             </span></span>
<span><span class="co">#&gt;  [85] ggrepel_0.9.6                    foreach_1.5.2                   </span></span>
<span><span class="co">#&gt;  [87] pillar_1.11.1                    stringr_1.5.2                   </span></span>
<span><span class="co">#&gt;  [89] yulab.utils_0.2.1                later_1.4.4                     </span></span>
<span><span class="co">#&gt;  [91] circlize_0.4.16                  splines_4.5.0                   </span></span>
<span><span class="co">#&gt;  [93] treeio_1.33.0                    lattice_0.22-7                  </span></span>
<span><span class="co">#&gt;  [95] bit_4.6.0                        tidyselect_1.2.1                </span></span>
<span><span class="co">#&gt;  [97] GO.db_3.21.0                     ComplexHeatmap_2.25.2           </span></span>
<span><span class="co">#&gt;  [99] Biostrings_2.77.0                knitr_1.50                      </span></span>
<span><span class="co">#&gt; [101] bookdown_0.44                    IRanges_2.43.0                  </span></span>
<span><span class="co">#&gt; [103] Seqinfo_0.99.2                   svglite_2.2.1                   </span></span>
<span><span class="co">#&gt; [105] shinydashboard_0.7.3             stats4_4.5.0                    </span></span>
<span><span class="co">#&gt; [107] xfun_0.53                        Biobase_2.69.0                  </span></span>
<span><span class="co">#&gt; [109] matrixStats_1.5.0                DT_0.34.0                       </span></span>
<span><span class="co">#&gt; [111] visNetwork_2.1.4                 stringi_1.8.7                   </span></span>
<span><span class="co">#&gt; [113] UCSC.utils_1.5.0                 lazyeval_0.2.2                  </span></span>
<span><span class="co">#&gt; [115] ggfun_0.2.0                      yaml_2.3.10                     </span></span>
<span><span class="co">#&gt; [117] kableExtra_1.4.0                 evaluate_1.0.5                  </span></span>
<span><span class="co">#&gt; [119] codetools_0.2-20                 tibble_3.3.0                    </span></span>
<span><span class="co">#&gt; [121] qvalue_2.41.0                    BiocManager_1.30.26             </span></span>
<span><span class="co">#&gt; [123] ggplotify_0.1.3                  cli_3.6.5                       </span></span>
<span><span class="co">#&gt; [125] arrow_21.0.0.1                   xtable_1.8-4                    </span></span>
<span><span class="co">#&gt; [127] systemfonts_1.2.3                jquerylib_0.1.4                 </span></span>
<span><span class="co">#&gt; [129] upsetjs_1.11.1                   EnhancedVolcano_1.27.0          </span></span>
<span><span class="co">#&gt; [131] Rcpp_1.1.0                       GenomeInfoDb_1.45.11            </span></span>
<span><span class="co">#&gt; [133] png_0.1-8                        parallel_4.5.0                  </span></span>
<span><span class="co">#&gt; [135] pkgdown_2.1.3                    ggplot2_4.0.0                   </span></span>
<span><span class="co">#&gt; [137] assertthat_0.2.1                 blob_1.2.4                      </span></span>
<span><span class="co">#&gt; [139] DOSE_4.3.0                       goat_1.1.3                      </span></span>
<span><span class="co">#&gt; [141] tidytree_0.4.6                   viridisLite_0.4.2               </span></span>
<span><span class="co">#&gt; [143] ggiraph_0.9.1                    scales_1.4.0                    </span></span>
<span><span class="co">#&gt; [145] openxlsx_4.2.8                   purrr_1.1.0                     </span></span>
<span><span class="co">#&gt; [147] crayon_1.5.3                     clisymbols_1.2.0                </span></span>
<span><span class="co">#&gt; [149] GetoptLong_1.0.5                 rlang_1.1.6                     </span></span>
<span><span class="co">#&gt; [151] cowplot_1.2.0                    fastmatch_1.1-6                 </span></span>
<span><span class="co">#&gt; [153] KEGGREST_1.49.1                  shinyjs_2.1.0</span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Maurits Unkel.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
